x-pack/plugins/security_solution/public/common/components/visualization_actions/use_save_to_library.test.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { renderHook, act } from '@testing-library/react-hooks';
+import { toMountPoint } from '@kbn/react-kibana-mount';
+import { useSaveToLibrary } from './use_save_to_library';
+import { useKibana } from '../../lib/kibana';
+import { useRedirectToDashboardFromLens } from './use_redirect_to_dashboard_from_lens';
+import { useGetSecuritySolutionUrl } from '../link_to';
+import { kpiHostMetricLensAttributes } from './lens_attributes/hosts/kpi_host_metric';
+
+jest.mock('../../lib/kibana', () => ({
+  useKibana: jest.fn(),
+}));
+
+jest.mock('./use_redirect_to_dashboard_from_lens', () => ({
+  useRedirectToDashboardFromLens: jest.fn(),
+}));
+
+jest.mock('../link_to', () => ({
+  useRedirectToDashboardFromLens: jest.fn(),
+  useGetSecuritySolutionUrl: jest.fn(),
+}));
+
+jest.mock('@kbn/react-kibana-mount', () => ({
+  toMountPoint: jest.fn(),
+}));
+
+describe('useSaveToLibrary hook', () => {
+  const mockStartServices = {
+    application: { capabilities: { visualize: { save: true } } },
+    lens: { SaveModalComponent: jest.fn() },
+  };
+
+  const mockGetSecuritySolutionUrl = jest.fn();
+  const mockRedirectTo = jest.fn();
+  const mockGetEditOrCreateDashboardPath = jest.fn().mockReturnValue('mockDashboardPath');
+
+  beforeEach(() => {
```

Comment:
We should call `jest.clearAllMocks();` inside the _beforeEach_, to make sure tests don't interfere with each other.